pkg/vsphere/toolbox/hgfs/server.go

```diff
@@ -440,9 +490,11 @@ func (s *Server) ReadV3(p *Packet) (interface{}, error) {
 
 	buf := make([]byte, req.RequiredSize)
 
-	n, err := file.ReadAt(buf, int64(req.Offset))
-	if err != nil {
-		if err != io.EOF || n <= 0 {
+	// Use ReadFull as Read() of an archive io.Pipe may return much smaller chunks,
+	// such as when we've read a tar header.
+	n, err := io.ReadFull(file, buf)
```

Comment:
Don't think we need ReadFrom/WriteTo in this case (in memory io.Pipe), but we could reuse the `buf` by attaching it to the archive struct (and hence the hgfs session).